Editor/Window/Utils/StateManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿using AmazonGameLift.Editor;
+using AmazonGameLiftPlugin.Core.ApiGatewayManagement;
+
+namespace Editor.Window.Utils
+{
+    public class StateManager
+    {
+        private State _state;
+        public CoreApi CoreApi { get; set; }
+        public GameLiftRequestAdapter RequestAdapter { get; set; }
+        public State State { get; set; }
+        public IAmazonGameLiftClientWrapper GameLiftWrapper { get; private set; }
+        
+        public IAmazonGameLiftClientFactory AmazonGameLiftClientFactory { get; set; }
+        
+        public void SetupClientFactory()
+        {
+            AmazonGameLiftClientFactory = new AmazonGameLiftClientFactory(CoreApi);
+        }
+        
+        public void SetupWrapper()
+        {
+            GameLiftWrapper = AmazonGameLiftClientFactory.Get(_state.SelectedProfile);
+        }
+
+        public void SetupRequestAdapter()
+        {
+            RequestAdapter = new GameLiftRequestAdapter(CoreApi, (AmazonGameLiftWrapper)GameLiftWrapper);
+        }
+    }
+    
+    public struct State
```

Comment:
I don't see the value in having all state variables in one struct. In my mind, there's very little use for passing this whole object around. Instead, most pages should just get the specific items they need. If there are multiple items that are related and used almost always together, then grouping those is fine

Reply:
Removed state and placed those into statemanager.